src/bridge/MigrationNFT.sol

```diff
@@ -3,38 +3,51 @@
 pragma solidity 0.8.23;
 
 import {ERC721} from "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Strings} from "openzeppelin-contracts/contracts/utils/Strings.sol";
 import {NODLMigration} from "./NODLMigration.sol";
 
 contract MigrationNFT is ERC721 {
+    using Strings for uint256;
+
     uint256 public nextTokenId;
-    uint256 public maxNFTs;
-    uint256 public minAmount;
+    uint256 public maxHolders;
```

Comment:
```suggestion
    uint256 public immutable maxHolders;
```

Reply:
Is it possible that we quickly issue all the NFTs for the lowest rank/level before we issue any high rank NFT?

Reply:
Yes that's a possibility, this is not supposed to be open forever but is instead a short lived incentive program